packages/toast/src/components/ToastComponent.tsx

```diff
@@ -0,0 +1,50 @@
+import {
+    EDSStyleSheet,
+    Icon,
+    Paper,
+    Typography,
+    useStyles,
+    useToken,
+} from "@equinor/mad-components";
+import React from "react";
+import { TouchableHighlight } from "react-native";
+import { ToastType } from "../types";
+import { getMaxToastWidth } from "../utils/getMaxToastWidth";
+import { getToastTokenForType } from "../utils/getToastTokenForType";
+
+export type ToastComponentProps = { text: string; type: ToastType; onPress?: () => void };
+export const ToastComponent = ({ text, type, onPress }: ToastComponentProps) => {
+    const styles = useStyles(theme, type);
+    const masterToken = useToken();
+    const { text: textColor } = getToastTokenForType(masterToken, type);
+
+    return (
+        <TouchableHighlight
```

Comment:
What is the reason `TouchableHighlight` was used here instead of the mad-components `PressableHighlight`?
[react native docs](https://reactnative.dev/docs/touchablehighlight) state that the Pressable API is more future proof, which was the reason for why we made `PressableHighlight`. It also handles styling for the pressed state

Reply:
Also, toasts that do not have the `onPress` method defined still behave as pressable components (darkens on touch). I think these toasts should not behave as pressable surfaces on user interaction

Reply:
I'll switch it to `PressableHighlight`.

When it comes to disabling the pressable effect, do you have any suggestions for how to do so? It looks like `react-native-toast-message` gives the component a callback function for 'simplicity', but that makes it hard to check if we have a real callback. That is the reason why my "disabled" logic doesn't work